supervisor/api/security.py

```diff
@@ -108,12 +109,30 @@
 class SecurityMiddleware(CoreSysAttributes):
     """Security middleware functions."""
 
-    def __init__(self, coresys):
+    def __init__(self, coresys: CoreSys):
         """Initialize security middleware."""
-        self.coresys = coresys
+        self.coresys: CoreSys = coresys
 
     @middleware
-    async def token_validation(self, request, handler):
+    async def system_validation(
+        self, request: Request, handler: RequestHandler
+    ) -> Response:
+        """Check if core is ready to response."""
+        if self.sys_core.state not in (
+            CoreState.STARTUP,
+            CoreState.RUNNING,
+            CoreState.FREEZE,
+        ):
+            return api_return_error(
```

Comment:
Should we return 425 here?
<https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/425>

Reply:
425 is marked as experimental.